src/renderer/wg_engine/tvgWgRenderer.cpp

```diff
@@ -476,17 +492,43 @@ bool WgRenderer::region(TVG_UNUSED RenderEffect* effect)
 }
 
 
-bool WgRenderer::render(TVG_UNUSED RenderCompositor* cmp, TVG_UNUSED const RenderEffect* effect, TVG_UNUSED bool direct)
+bool WgRenderer::render(RenderCompositor* cmp, const RenderEffect* effect, TVG_UNUSED bool direct)
 {
-    TVGLOG("WG_ENGINE", "SceneEffect(%d) is not supported", (int)effect->type);
+    // we must to end current render pass to resolve ms texture before effect
+    mCompositor.endRenderPass();
+    // expand current aabb
+    WgCompose* comp = (WgCompose *)cmp;
+    comp->aabb.x -= WG_GAUSSIAN_KERNEL_SIZE;
+    comp->aabb.y -= WG_GAUSSIAN_KERNEL_SIZE;
+    comp->aabb.w += WG_GAUSSIAN_KERNEL_SIZE * 2;
+    comp->aabb.h += WG_GAUSSIAN_KERNEL_SIZE * 2;
+    // get current render target
+    WgRenderStorage* dst = mRenderStorageStack.last();
+
+    // apply effect
+    switch (effect->type) {
+        case SceneEffect::GaussianBlur: return mCompositor.gaussianBlur(mContext, dst, static_cast<const RenderEffectGaussianBlur*>(effect));
+        case SceneEffect::DropShadow:
+        case SceneEffect::Fill:
+        case SceneEffect::Tint:
+        case SceneEffect::Tritone: return false;
+        default: return false;
+    }
     return false;
 }
 
 
-void WgRenderer::dispose(TVG_UNUSED RenderEffect* effect)
+void WgRenderer::dispose(RenderEffect* effect)
 {
-    //TODO: dispose the effect
-}
+    // dispose gaussian blur data
+    if (effect->type == SceneEffect::GaussianBlur) {
+        auto gaussianBlur = (RenderEffectGaussianBlur*)effect;
+        auto renderDataGaussian = (WgRenderDataGaussian *)gaussianBlur->rd;
+        if (renderDataGaussian)
+            mRenderDataGaussianPool.free(mContext, renderDataGaussian);
```

Comment:
suggestion handling nullptr in free() method,  you can make this single line:
`mRenderDataGaussianPool.free(mContext, WgRenderDataGaussian*) gaussianBlur->rd);`